packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -205,17 +226,98 @@ export const SampleFinderSection: FC<Props> = memo(props => {
             });
 
             onFilterClose();
-            updateFilters(filterChangeCounter + 1, newFilterCards);
+            updateFilters(filterChangeCounter + 1, newFilterCards, !currentView?.entityId, true);
 
             api.query.incrementClientSideMetricCount(SAMPLE_FILTER_METRIC_AREA, 'filterModalApply');
         },
-        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType]
+        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType, cardDirty, currentView]
+    );
+
+    const loadSearch = useCallback(
+        async (view: FinderReport) => {
+            let cardJson = null;
+
+            if (view.isSession) cardJson = sessionStorage.getItem(getLocalStorageKey());
+            else if (view.reportId) {
+                try {
+                    cardJson = await loadFinderSearch(view);
+                } catch (error) {
+                    console.error(error);
+                    return;
+                }
+            }
+            if (!cardJson) return;
+
+            const finderSessionData = searchFiltersFromJson(cardJson);
+            const newFilters = finderSessionData.filters;
+            if (!newFilters) return;
+
+            updateFilters(filterChangeCounter + 1, newFilters, false, view.isSession);
+            setShowSaveViewDialog(false);
+            setCurrentView(view);
+        },
+        [filterChangeCounter]
+    );
+
+    const onSaveComplete = useCallback(
+        (view: FinderReport) => {
+            setShowSaveViewDialog(false);
+            setViewDirty(false);
+            setCurrentView(view);
+            setSavedViewChangeCounter(savedViewChangeCounter + 1);
+        },
+        [savedViewChangeCounter]
+    );
+
+    const searchViewJson = useMemo(() => {
+        return JSON.stringify({
+            filters: getSearchFilterObjs(filters),
+        });
+    }, [filters]);
+
+    const saveSearch = useCallback(
+        async (saveCurrentName?: boolean) => {
+            try {
+                if (saveCurrentName) {
+                    await saveFinderSearch(currentView, searchViewJson, saveCurrentName);
+                    setViewDirty(false);
+                } else {
+                    setShowSaveViewDialog(true);
+                }
+            } catch (error) {
+                console.error(error);
```

Comment:
do you think we need to show this error to the user in some way if the save of their search failed for some reason?